src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -217,4 +292,136 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String queryId() {
         return NAME;
     }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        String field;
+        FieldMapper mapper = parseContext.fieldMapper(fieldName);
+        if (mapper != null) {
+            field = mapper.names().indexName();
+        } else {
+            field = fieldName;
+        }
+
+        Analyzer analyzerObj;
+        if (analyzer == null) {
+            if (mapper != null) {
+                analyzerObj = parseContext.getSearchAnalyzer(mapper);
+            } else {
+                analyzerObj = parseContext.mapperService().searchAnalyzer();
+            }
```

Comment:
I had a closer look too, but its a bit confusing. Worse, there are changes to the mappers coming in from master where we now have `mapper.fieldType().searchAnalyzer()` instead of `mapper.searchAnalyzer`.

By looking at the implementation in QueryParseContext think the second if-clause (parseContext.getSearchAnalyzer(mapper)) takes care of all cases where mapper is not null, already covering the first if-clause in the original code. I think in the original code, the third if clause is redundant when mapper is not null, because it only tries calling the same method again that parseContext.getSearchAnalyzer(mapper) does internally.
So I think the simplification works here, although I think that `analyzerObj` might still be `null`afterwards. Maybe the check for (analyzerObj == null) should be later, outside of all the if/else cases, just to make sure.